rust-runtime/aws-smithy-runtime-api/src/client/endpoints.rs

```diff
@@ -28,8 +34,77 @@ impl StaticUriEndpointResolver {
 }
 
 impl EndpointResolver for StaticUriEndpointResolver {
-    fn resolve_and_apply_endpoint(&self, request: &mut HttpRequest) -> Result<(), BoxError> {
+    fn resolve_and_apply_endpoint(
+        &self,
+        _params: &EndpointResolverParams,
+        _endpoint_prefix: Option<&EndpointPrefix>,
+        request: &mut HttpRequest,
+    ) -> Result<(), BoxError> {
         apply_endpoint(request.uri_mut(), &self.endpoint, None)?;
         Ok(())
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct DefaultEndpointResolver<Params> {
```

Comment:
This should be in the `aws-smithy-runtime` crate rather than `-api` since it is a trait implementation rather than a trait.

Reply:
Makes sense. Addressed in 8219785.